samples/snippets/detect_intent_stream.py

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python
+
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""DialogFlow API Detect Intent Python sample with audio files processed as an audio stream.
+
+Examples:
+  python detect_intent_stream.py -h
+  python detect_intent_stream.py --agent AGENT \
+  --session-id SESSION_ID --audio-file-path resources/hello.wav
+"""
+
+import argparse
+import uuid
+
+from google.cloud.dialogflowcx_v3beta1.services.sessions import SessionsClient
+from google.cloud.dialogflowcx_v3beta1.types import audio_config
+from google.cloud.dialogflowcx_v3beta1.types import session
+
+
+# [START dialogflow_detect_intent_stream]
+def detect_intent_stream(agent, session_id, audio_file_path, language_code):
+    """Returns the result of detect intent with streaming audio as input.
+
+    Using the same `session_id` between requests allows continuation
+    of the conversation."""
+    session_client = SessionsClient()
+    session_path = "{}/sessions/{}".format(agent, session_id)
+    print("Session path: {}\n".format(session_path))
+
+    input_audio_config = audio_config.InputAudioConfig(
+        audio_encoding=audio_config.AudioEncoding.AUDIO_ENCODING_LINEAR_16,
+        sample_rate_hertz=24000,
+    )
+
+    def request_generator():
+        audio_input = session.AudioInput(config=input_audio_config)
+        query_input = session.QueryInput(audio=audio_input, language_code=language_code)
+
+        # The first request contains the configuration.
+        yield session.StreamingDetectIntentRequest(
+            session=session_path, query_input=query_input
+        )
+
+        # Here we are reading small chunks of audio data from a local
+        # audio file.  In practice these chunks should come from
+        # an audio input device.
+        with open(audio_file_path, "rb") as audio_file:
+            while True:
+                chunk = audio_file.read(4096)
+                if not chunk:
+                    break
+                # The later requests contains audio data.
+                audio_input = session.AudioInput(audio=chunk)
+                query_input = session.QueryInput(audio=audio_input)
+                yield session.StreamingDetectIntentRequest(query_input=query_input)
+
+    requests = request_generator()
+    responses = session_client.streaming_detect_intent(requests=requests)
```

Comment:
nit: inline this argument
```suggestion
    responses = session_client.streaming_detect_intent(requests=request_generator())
```